### ISSUE TITLE
mypy fails with "incompatible type"

### ISSUE THREAD

**Issue by YorikSar** [cript0nauta/pynixify#52]
Trying to install with latest nixpkgs fails with:

```
running install tests
pynixify/nixpkgs_sources.py:121: error:(B Argument 2 to (B"create_subprocess_exec"(B has incompatible type (B"*List[Sequence[str]]"(B; expected (B"Union[str, bytes, _PathLike[str], _PathLike[bytes]]"(B(B
Found 1 error in 1 file (checked 19 source files)(B
builder for '/nix/store/ln17xrkz885vk7xl0p6k13xz3hax8rc7-python3.8-pynixify-0.1.drv' failed with exit code 1
error: build of '/nix/store/ln17xrkz885vk7xl0p6k13xz3hax8rc7-python3.8-pynixify-0.1.drv' failed
```

To reproduce - run `nix-build .` in current Git master with latest nixpkgs.

**Comment by cript0nauta** [comment]
Thanks for reporting. There seems to be a problem when building pynixify using mypy 0.790, which is the version used in nixpkgs unstable. In addition to this, the tests are also failing when using nixpkgs unstable, probably due to an upgrade in a dependency.

I'll be looking at both problems and try to fix them soon.

**Comment by cript0nauta** [comment]
This will take longer than what I initially expected. Nixpkgs unstable has upgraded from setuptools 47.3.1 to 50.3.1, which introduces breaking changes. Even if I fix the build of pynixify, you'll experience many errors using the examples in the README.

For now, I'd recommend to use nixpkgs stable (channel `nixos-20.09` should work).

**Comment by cript0nauta** [comment]
Support for nixpkgs-unstable is blocked until the regression in https://github.com/NixOS/nixpkgs/issues/108525 is fixed.

**Comment by cript0nauta** [comment]
Fixed in 127d07263dc5460929af932c7c87508434aa4f27. Also, nixos-unstable and nixos-21.05 channels are now working ok.
